ansible_navigator/configuration/application_configuration.py

```diff
@@ -0,0 +1,199 @@
+from .definitions import Config
+from .definitions import CliParameters
+from .definitions import SubCommand
+from .definitions import Entry
+from .definitions import EntryValue
+
+from ..utils import oxfordcomma
+
+
+def abs_user_path(fpath):
+    """Resolve a path"""
+    return os.path.abspath(os.path.expanduser(fpath))
+
+
+def generate_editor_command():
+    """Generate a default for editor_command if EDITOR is set"""
+    if "EDITOR" in os.environ:
+        command = "%s {filename}" % os.environ.get("EDITOR")
+    else:
+        command = "vi +{line_number} {filename}"
+    return command
+
+
+PLUGIN_TYPES = (
+    "become",
+    "cache",
+    "callback",
+    "cliconf",
+    "connection",
+    "httpapi",
+    "inventory",
+    "lookup",
+    "module",
+    "netconf",
+    "shell",
+    "strategy",
+    "vars",
+)
+
+ApplicationConfiguration = Config(
+    application_name="ansible-navigator",
+    post_processor=ApplicationPostProcessor(),
+    subcommands=[
+        SubCommand(name="collections", description="Explore available collections"),
+        SubCommand(name="config", description="Explore the current ansible configuration"),
+        SubCommand(name="doc", description="Review documentation for a module or plugin"),
+        SubCommand(name="inventory", description="Explore an inventory"),
+        SubCommand(name="load", description="Explore a playbook artifact"),
+        SubCommand(name="run", description="Run a playbook"),
+    ],
+    entries=[
+        Entry(
+            name="app",
+            description="Placeholder for subcommand name",
+            subcommand_value=True,
+            value=EntryValue(default="welcome"),
+        ),
+        Entry(
+            name="cmdline",
+            description="Placeholder for argparse remainder",
+            value=EntryValue(default=[]),
+        ),
+        Entry(
+            name="container_engine",
+            choices=["podman", "docker"],
+            cli_parameters=CliParameters(short="--ce"),
+            description="Specify the container engine",
+            value=EntryValue(default="podman"),
+        ),
+        Entry(
+            name="editor_command",
+            cli_parameters=CliParameters(short="--ecmd"),
+            description="Specify the editor comamnd",
+            settings_file_path_override="editor.command",
+            value=EntryValue(default=generate_editor_command()),
+        ),
+        Entry(
+            name="editor_console",
+            choices=[True, False],
+            cli_parameters=CliParameters(short="--econ"),
+            description="Specify if the editor is console based",
+            settings_file_path_override="editor.console",
+            value=EntryValue(default=True),
+        ),
+        Entry(
+            name="execution_environment",
+            choices=[True, False],
+            cli_parameters=CliParameters(short="--ee"),
+            description="Enable the use of an execution environment",
```

Comment:
We can have a top section for EE parameters `execution_environment` as there can be multiple options related to EE?
`execution_environment`, `execution_environment_image`, `container_engine`, `container_mount_path`

Reply:
I'm assuming you mean in the settings file, so we can add execution-environment as a parent key and nest the related params

Reply:
`  execution-environment:
    container-engine: podman
    enabled: False
    environment-variables:
      pass:
        - ONE
        - TWO
        - THREE
      set:
        KEY1: VALUE1
        KEY2: VALUE2
        KEY3: VALUE3
    image: test_image`

Reply:
set and pass environment variables will be moved under a the same key since they are specific to the ee